### PR TITLE
AArch64: enable fc tests on ARM CI

### PR DESCRIPTION
**Pull request by Pennyzct**
Hi~ guys
I've been trying to enable FC tests on ARM CI lately.
All tests work well in my local machine and I may need to do the following twist on upstream existing configuration.

- **configure_docker: add -s option for configuring storage driver**
When running with firecracker, docker is configured by `/etc/docker/daemon.json`, which is inconsistent with the original installation in [install_docker()](https://github.com/kata-containers/tests/blob/master/.ci/setup.sh#L49). They are using systemd service file to configure.
Let's stick with one method, and add new `-s` option to configure storage driver.

- **FC: speed up installing firecracker/jailer**
Since we only use `firecracker/jailer` in stable version, we could just download binary from their release page.Therefore we could save the time and bandwidth.
For now, fc release page only provides binaries in two archs: `aarch64` and `x86_64`.

### CODE REVIEW

File: cmd/container-manager/manage_ctr_mgr.sh [hunk 1]
```diff
@@ -28,11 +28,12 @@ Options:
 	-f                         : Force action. It will replace any installation
                                      or configuration that you may have.
 	-h                         : Help, show this information.
-	-r <runtime>               : Supported runtimes: runc and cc-runtime.
```
Comment (jodh-intel) [inline cmd/container-manager/manage_ctr_mgr.sh]: Ha! :smile:

File: cmd/container-manager/manage_ctr_mgr.sh [hunk 2]
```diff
@@ -60,7 +61,7 @@ log_message(){
 }
 
 parse_subcommand_options(){
-	while getopts ":fr:t:" opt; do
+	while getopts ":fr:t:s:" opt; do
```
Comment (jodh-intel) [inline cmd/container-manager/manage_ctr_mgr.sh]: Nit: Not sorted alphabetically.

File: cmd/container-manager/manage_ctr_mgr.sh [hunk 3]
```diff
@@ -71,6 +72,9 @@ parse_subcommand_options(){
 			t)
 				tag="${OPTARG}"
 				;;
+			s)
```
Comment (jodh-intel) [inline cmd/container-manager/manage_ctr_mgr.sh]: Again, this should come before the `t)` block above :)

File: .ci/install_firecracker.sh [hunk 4]
```diff
@@ -24,73 +24,31 @@ if [ "$docker_version" != "18.06" ]; then
...
-	go get -d "${firecracker_repo}" || true
-	# Checkout to specific version
-	pushd "${GOPATH}/src/${firecracker_repo}"
-	git checkout "tags/${firecracker_version}"
-	./tools/devtool --unattended build --release
-	sudo install ${GOPATH}/src/${firecracker_repo}/build/cargo_target/${arch}-*-linux-musl/release/firecracker /usr/bin/
-	sudo install ${GOPATH}/src/${firecracker_repo}/build/cargo_target/${arch}-*-linux-musl/release/jailer /usr/bin/
-	popd
+	# Download firecracker and jailer
+	curl -fsL ${firecracker_repo}/releases/download/v${firecracker_version}/firecracker-v${firecracker_version}-${arch} -o /usr/bin/firecracker
```
Comment (jodh-intel) [inline .ci/install_firecracker.sh]: Please split these commands into two parts:

1) Download as the current user.
2) Run `sudo install` to install them.

File: cmd/container-manager/manage_ctr_mgr.sh (hunk 1 above)
Comment (chavafg) [reply cmd/container-manager/manage_ctr_mgr.sh]: lol, good catch :)

**Comment by Pennyzct** [comment]
Hi~ @jodh-intel @chavafg 
adressed all suggestions~~~~ plz take a review. ;)

**Comment by jodh-intel** [comment]
F31 CI failed with `EPERM`, so restarted to see if issue is transitory:

```
07:47:21 podman attach to the latest container		7.047083
07:47:22 "unlinkat /tmp/podman/imagecachedir/vfs/dir/083c11ecaa71f30cd96462da54e18aa870d7cddfa199f9ff31ed8d9913d56b40/root/original-ks.cfg: permission denied"
07:47:22 
07:47:22 
07:47:22 Summarizing 1 Failure:
07:47:22 
07:47:22 [Fail] Podman exec [It] podman exec simple working directory test 
```

Also restarted the OpenSUSE CI which failed with:

```
7:54:37 bash sanity/check_sanity.sh
07:54:37 5dc10e79ee045f4691c2cfbb160b731a82c086b57086daef209586cd27c1873b
07:54:37 ERROR: 1 pods left and found at /run/vc/sbs
```

**Comment by Pennyzct** [comment]
x86_64 FC failed on the following error:
```
21:52:52 info docker info 
21:52:52   should has a runtime information
21:52:52   /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:23
21:52:52 Running command '/usr/bin/docker [docker info]'
21:52:52 [docker info]
21:52:52 Timeout: 120 seconds
21:52:52 Exit Code: 0
21:52:52 Stdout: Containers: 0
21:52:52  Running: 0
21:52:52  Paused: 0
21:52:52  Stopped: 0
21:52:52 Images: 7
21:52:52 Server Version: 18.06.3-ce
21:52:52 Storage Driver: overlay2
21:52:52  Backing Filesystem: extfs
21:52:52  Supports d_type: true
21:52:52  Native Overlay Diff: false
21:52:52 Logging Driver: json-file
21:52:52 Cgroup Driver: cgroupfs
21:52:52 Plugins:
21:52:52  Volume: local
21:52:52  Network: bridge host macvlan null overlay
21:52:52  Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
21:52:52 Swarm: inactive
21:52:52 Runtimes: runc
21:52:52 Default Runtime: runc
21:52:52 Init Binary: docker-init
21:52:52 containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
21:52:52 runc version: a592beb5bc4c4092b1b1bac971afed27687340c5
21:52:52 init version: fec3683
21:52:52 Security Options:
21:52:52  apparmor
21:52:52  seccomp
21:52:52   Profile: default
21:52:52 Kernel Version: 5.0.0-1032-azure
21:52:52 Operating System: Ubuntu 18.04.4 LTS
21:52:52 OSType: linux
21:52:52 Architecture: x86_64
21:52:52 CPUs: 4
21:52:52 Total Memory: 15.64GiB
21:52:52 Name: ubuntu1804-azure479d10
21:52:52 ID: TSPQ:3HYP:VVSI:HGDU:VZJA:Y2X2:E5HM:T5WR:6FCQ:DHN2:Y2SX:CI6G
21:52:52 Docker Root Dir: /var/lib/docker
21:52:52 Debug Mode (client): false
21:52:52 Debug Mode (server): false
21:52:52 Registry: https://index.docker.io/v1/
21:52:52 Labels:
21:52:52 Experimental: false
21:52:52 Insecure Registries:
21:52:52  127.0.0.0/8
21:52:52 Live Restore Enabled: false
21:52:52 
21:52:52 
21:52:52 Stderr: WARNING: No swap limit support
21:52:52 
21:52:52 
21:52:52 • Failure [0.083 seconds]
21:52:52 info
21:52:52 /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:16
21:52:52   docker info
21:52:52   /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:22
21:52:52     should has a runtime information [It]
21:52:52     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:23
21:52:52 
21:52:52     Expected
21:52:52         <bool>: false
21:52:52     to be true
21:52:52 
21:52:52     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:28
21:52:52 ------------------------------
21:52:52 SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
21:52:52 
21:52:52 Summarizing 1 Failure:
21:52:52 
21:52:52 [Fail] info docker info [It] should has a runtime information 
21:52:52 /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/info_test.go:28
21:52:52 
21:52:52 Ran 2 of 250 Specs in 41.594 seconds
21:52:52 FAIL! -- 1 Passed | 1 Failed | 0 Pending | 248 Skipped
```
`kata-runtim` has not been configured as a new runtime in docker. I've raised a new commit to fix this~~

File: .ci/install_runtime.sh [hunk 5]
```diff
@@ -116,12 +116,12 @@ fi
 
 if [ "${TEST_CGROUPSV2}" == "false" ]; then
 	case "${KATA_HYPERVISOR}" in
-		"cloud-hypervisor" | "qemu")
-			echo "Add kata-runtime as a new/default Docker runtime."
```
Comment (Pennyzct) [inline .ci/install_runtime.sh]: For clarification, `manage_ctr_mgr.sh" docker configure -r kata-runtime -f` will not override default runtime as this new runtime(here, `kata-runtime`), the default one will still be `runc`, except it is set for `docker swarm`
see https://github.com/kata-containers/tests/blob/master/cmd/container-manager/manage_ctr_mgr.sh#L299

**Comment by chavafg** [comment]
hmm, so now all failed because https://github.com/kata-containers/runtime/pull/2505 needs to be fixed.

**Comment by Pennyzct** [comment]
Hi guys
After completing changes on https://github.com/kata-containers/runtime/pull/2505, CI is happy now~~~

**Pull request merged**